### ISSUE TITLE
Round doesn't change when proposer is down

### ISSUE THREAD

**Issue by aneequesafdar** [clearmatics/autonity#156]
When we run `local_test_network` with 4 nodes only where the proposer is not started than consensus doesn't move to new round.

Here is a log from one of the node:

```
DEBUG[07-01|14:23:52.215|consensus/tendermint/core/core.go:280]       Starting new Round                       Height=1 Round=0
INFO [07-01|14:23:52.215|consensus/tendermint/backend/engine.go:352]  Soma Contract Deployer                   Address=0x0000000000000000000000000000000000000000
INFO [07-01|14:23:52.219|consensus/tendermint/backend/contract.go:72] Deployed Soma Governance Contract        Address=0xBd770416a3345F91E4B34576cb804a576fa48EB1
DEBUG[07-01|14:23:52.220|p2p/server.go:786]                           Adding p2p peer                          name=Autonity/v1.0.0-alph... addr=172.25.0.14:30303 peers=1
DEBUG[07-01|14:23:52.221|eth/handler.go:310]                          Ethereum peer connected                  id=5b68cfe243a23a70 conn=staticdial name=Autonity/v1.0.0-alpha-63b26399/linux-amd64/go1.11.11
DEBUG[07-01|14:23:52.221|core/types/nodes.go:39]                      performing                               node=enode://1f207dfb3bcbbd338fbc991ec13e40d204b58fe7275cea48cfeb53c2c24e1071e1b4ef2959325fe48a5893de8ff37c73a24a412f367e505e5dec832813da546a@172.25.0.12:30303
DEBUG[07-01|14:23:52.222|core/types/nodes.go:39]                      performing                               node=enode://3ce6c053cb563bfd94f4e0e248510a07ccee1bc836c9784da1816dba4b10564e7be1ba42e0bd8d73c8f6274f8e9878dc13814adb381c823264265c06048b4b59@172.25.0.15:30303
DEBUG[07-01|14:23:52.222|core/types/nodes.go:39]                      performing                               node=enode://d73b857969c86415c0c000371bcebd9ed3cca6c376032b3f65e58e9e2b79276fbc6f59eb1e22fcd6356ab95f42a666f70afd4985933bd8f3e05beb1a2bf8fdde@172.25.0.11:30303
DEBUG[07-01|14:23:52.222|core/types/nodes.go:39]                      performing                               node=enode://e766ac390e2d99b559aef773c3656fa8d50df2310496ac26ca6c3fc84e21dabb8a0162cc8e34f938d45e0a8ed04955f8ddf1c380182f8ef17a3f08885064505f@172.25.0.13:30303
DEBUG[07-01|14:23:52.222|core/types/nodes.go:39]                      performing                               node=enode://438a5c2cd8fdc2ecbc508bf7362e41c0f0c3754ba1d3267127a3756324caf45e6546b02140e2144b205aeb372c96c5df9641485f721dc7c5b27eb9e35f5d887b@172.25.0.14:30303
INFO [07-01|14:23:52.224|core/network_permissioning.go:134]           Deployed Glienicke Contract              Address=0x522B3294E6d06aA25Ad0f1B8891242E335D3B459
INFO [07-01|14:23:52.225|miner/worker.go:990]                         Commit new mining work                   number=1 sealhash=d00b10…a87403 uncles=0 txs=0 gas=0 fees=0 elapsed=10.977ms
INFO [07-01|14:23:52.226|miner/worker.go:540]                         New block Seal request                   hash=d00b10…a87403
DEBUG[07-01|14:23:52.227|consensus/tendermint/core/handler.go:78]     Started handling tendermint.NewUnminedBlockEvent 
DEBUG[07-01|14:23:52.227|consensus/tendermint/core/unmined_block.go:89] NewUnminedBlockEvent: Received           from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f type="New Unmined Block" hash=777255…42b525 currentHeight=1 currentRound=0 currentStep=StepAcceptProposal currentProposer=false msgHeight=1 isNilMsg=false
DEBUG[07-01|14:23:52.228|consensus/tendermint/core/handler.go:93]       Finished handling tendermint.NewUnminedBlockEvent 
DEBUG[07-01|14:23:52.228|consensus/tendermint/core/core.go:299]         Scheduled Propose Timeout                Timeout Duration=10s
DEBUG[07-01|14:23:53.154|p2p/server.go:786]                             Adding p2p peer                          name=Autonity/v1.0.0-alph...                              addr=172.25.0.13:30303 peers=2
DEBUG[07-01|14:23:53.154|eth/handler.go:310]                            Ethereum peer connected                  id=02619de2661819c5 conn=staticdial name=Autonity/v1.0.0-alpha-63b26399/linux-amd64/go1.11.11
DEBUG[07-01|14:23:53.158|eth/handler.go:321]                            Ethereum handshake failed                id=02619de2661819c5 conn=staticdial err=EOF
DEBUG[07-01|14:23:53.158|p2p/server.go:804]                             Removing p2p peer                        id=02619de2661819c5 conn=staticdial duration=4.077ms peers=1 req=true err="already connected"
DEBUG[07-01|14:23:53.869|p2p/server.go:786]                             Adding p2p peer                          name=Autonity/v1.0.0-alph...                              addr=172.25.0.12:53326 peers=2
DEBUG[07-01|14:23:53.869|eth/handler.go:310]                            Ethereum peer connected                  id=fd3b8aca297dbea5 conn=inbound    name=Autonity/v1.0.0-alpha-63b26399/linux-amd64/go1.11.11
DEBUG[07-01|14:23:53.870|eth/downloader/downloader.go:1746]             Relaxed downloader QoS values            rtt=20s confidence=0.500 ttl=1m0s
DEBUG[07-01|14:23:54.273|p2p/nat/nat.go:109]                            Couldn't add port mapping                proto=tcp extport=30303 intport=30303 interface="UPnP or NAT-PMP" err="no UPnP or NAT-PMP router discovered"
DEBUG[07-01|14:23:55.259|p2p/dial.go:317]                               Can't resolve node                       id=0x8dcb80         err="discovery is disabled"
DEBUG[07-01|14:24:00.911|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:24:00.915|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0xc443C6c6AE98F5110702921138D840e77dA67702
DEBUG[07-01|14:24:00.916|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
DEBUG[07-01|14:24:02.228|consensus/tendermint/core/timeout.go:48]       TimeoutEvent(Propose): Sent              from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f type=Propose             currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepAcceptProposal msgStep=0
DEBUG[07-01|14:24:02.228|consensus/tendermint/core/handler.go:154]      Started handling timeoutEvent 
DEBUG[07-01|14:24:02.228|consensus/tendermint/core/timeout.go:48]       TimeoutEvent(Propose): Received          from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f type=Propose             currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepAcceptProposal msgStep=0
DEBUG[07-01|14:24:02.228|consensus/tendermint/core/prevote.go:127]      MessageEvent(Prevote): Sent              from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f to=broadcast currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepAcceptProposal isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=true  hash=000000…000000 type=Prevote             totalVotes=0 totalNilVotes=0 quorumReject=false totalNonNilVotes=0 quorumAccept=false
DEBUG[07-01|14:24:02.228|consensus/tendermint/core/core.go:187]         broadcasting                             step=StepAcceptProposal msg="{Code: 1, Address: 0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f}"
DEBUG[07-01|14:24:02.280|consensus/tendermint/core/handler.go:163]      Finished handling timeoutEvent 
DEBUG[07-01|14:24:02.280|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:24:02.281|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f
DEBUG[07-01|14:24:02.281|consensus/tendermint/core/prevote.go:127]      MessageEvent(Prevote): Received          from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepProposeDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=true  hash=000000…000000 type=Prevote             totalVotes=1 totalNilVotes=1 quorumReject=false totalNonNilVotes=0 quorumAccept=false
DEBUG[07-01|14:24:02.281|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
DEBUG[07-01|14:24:03.289|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:24:03.289|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D
DEBUG[07-01|14:24:03.290|consensus/tendermint/core/prevote.go:127]      MessageEvent(Prevote): Received          from=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepProposeDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=true  hash=000000…000000 type=Prevote             totalVotes=2 totalNilVotes=2 quorumReject=false totalNonNilVotes=0 quorumAccept=false
DEBUG[07-01|14:24:03.290|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
DEBUG[07-01|14:24:03.966|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:24:03.966|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5
DEBUG[07-01|14:24:03.966|consensus/tendermint/core/prevote.go:127]      MessageEvent(Prevote): Received          from=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepProposeDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=true  hash=000000…000000 type=Prevote             totalVotes=3 totalNilVotes=3 quorumReject=false totalNonNilVotes=0 quorumAccept=false
DEBUG[07-01|14:24:03.966|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
DEBUG[07-01|14:24:04.056|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:24:04.057|consensus/tendermint/core/handler.go:258]      tendermint.MessageEvent: PRECOMMIT       address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0xc443C6c6AE98F5110702921138D840e77dA67702
DEBUG[07-01|14:24:04.057|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0xc443C6c6AE98F5110702921138D840e77dA67702 to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=1 msgHeight=1 currentRound=0 msgRound=0 currentStep=StepProposeDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=true  hash=000000…000000 type=Precommit           totalVotes=1 totalNilVotes=1 quorumReject=false totalNonNilVotes=0 quorumAccept=false
DEBUG[07-01|14:24:04.057|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
```

**Comment by aneequesafdar** [comment]
So it seems that the messaging layer is not the problem, rather the handling of the messages is not working properly. In the following log node 1 receives prevotes but does nothing with them, here is a log:

```
DEBUG[07-01|14:48:13.915|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:48:13.916|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5
DEBUG[07-01|14:48:13.916|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
WARN [07-01|14:48:13.916|consensus/tendermint/backend/backend.go:341]   worker loop. resend to connected peers   msg=0x3c564785862a8decc04549b2f526ed674874c56402be99fbc773634383ab74b8 peers="peers 3: 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
WARN [07-01|14:48:13.916|consensus/tendermint/backend/backend.go:241]   inner sender loop. message sent earlier  peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x3c564785862a8decc04549b2f526ed674874c56402be99fbc773634383ab74b8
WARN [07-01|14:48:13.966|consensus/tendermint/backend/backend.go:269]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x3c564785862a8decc04549b2f526ed674874c56402be99fbc773634383ab74b8
WARN [07-01|14:48:13.966|consensus/tendermint/backend/backend.go:269]   inner sender loop. success               try=1 peer=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D msg=0x3c564785862a8decc04549b2f526ed674874c56402be99fbc773634383ab74b8
DEBUG[07-01|14:48:14.946|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:48:14.946|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D
DEBUG[07-01|14:48:14.946|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
WARN [07-01|14:48:14.947|consensus/tendermint/backend/backend.go:341]   worker loop. resend to connected peers   msg=0x7cc73f2d5d6061cd25c4cd380bfe8628d13fe9e6a5e8df8c7bd7b950ec661563 peers="peers 3: 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
WARN [07-01|14:48:14.947|consensus/tendermint/backend/backend.go:241]   inner sender loop. message sent earlier  peer=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D msg=0x7cc73f2d5d6061cd25c4cd380bfe8628d13fe9e6a5e8df8c7bd7b950ec661563
WARN [07-01|14:48:14.997|consensus/tendermint/backend/backend.go:269]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x7cc73f2d5d6061cd25c4cd380bfe8628d13fe9e6a5e8df8c7bd7b950ec661563
WARN [07-01|14:48:14.997|consensus/tendermint/backend/backend.go:269]   inner sender loop. success               try=1 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x7cc73f2d5d6061cd25c4cd380bfe8628d13fe9e6a5e8df8c7bd7b950ec661563
DEBUG[07-01|14:48:15.878|consensus/tendermint/core/handler.go:124]      Started handling tendermint.MessageEvent 
DEBUG[07-01|14:48:15.880|consensus/tendermint/core/handler.go:255]      tendermint.MessageEvent: PREVOTE         address=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f from=0xc443C6c6AE98F5110702921138D840e77dA67702
DEBUG[07-01|14:48:15.880|consensus/tendermint/core/handler.go:129]      Finished handling tendermint.MessageEvent 
```

**Comment by aneequesafdar** [comment]
closed via #159
